et_replay/comm/commsTraceParser.py

```diff
@@ -0,0 +1,205 @@
+    "Tensor(c10::BFloat16)": "bfloat16",
+    "Tensor(unsigned char)": "char",
+    "Tensor(signed char)": "char",
+}
+
+
+def parseTrace(
+    in_trace: List, trace_type: str, trace_file_path: str, target_rank: int, total_ranks: int
+) -> List:
+    """
+    Parse trace files to be compatible with PARAM replay-mode.
+    Currently supports: Chakra host execution trace.
+
+    Args:
+        in_trace: Trace file to be parsed.
+        trace_type: Trace type to be parsed with
+        trace_file_path: Path of input trace file being loaded.
+        target_rank: The current rank of the device.
+        total_ranks: Total number of ranks.
+    Returns:
+        parsed_trace: Parsed trace that is compatible with PARAM replay-mode.
+    """
+
+    if trace_type == "et":  # Execution Trace (e.g. Chakra host execution trace)
+        parsed_trace = _parseExecutionTrace(
+            ExecutionTrace(in_trace), target_rank, total_ranks
+        )
+    else:
+        raise ValueError(f"Specified trace type {trace_type} to {trace_file_path} is not supported. \
+Please check supported types with '--help'")
+
+    return parsed_trace
+
+
+def _getTensorInfoFromPyTorchETEntry(
+    tensor_container: List, container_type: str
+) -> Tuple[int, int, str]:
+    """
+    Extract message size, tensor count, type from PyTorch ET entry inputs/outputs field.
+    NOTE: This format can be changed at anytime. TODO: When an extract/parsing tool is available in ATC, switch to it.
+    """
+    list_count = container_type.count("GenericList")
+    tensors = []
+    if list_count == 2:
+        # GenericList[GenericList[Tensor(), Tensor()]]
+        tensors = tensor_container[0][0]
+        dtype = container_type.replace("GenericList[", "").split(",", 1)[0]
+    elif list_count == 1:
+        # GenericList[Tensor()]
+        tensors = tensor_container[0]
+        dtype = container_type.replace("GenericList[", "").replace("]", "")
+    else:
+        tensors.append(tensor_container[0])
+        dtype = container_type
+
+    msg_size = 0
+    for tensor in tensors:
+        msg_size += tensor[3]
+
+    return msg_size, dtype
+
+
+def _parseExecutionTrace(
+    in_trace: ExecutionTrace, target_rank: int, total_ranks: int
+) -> List:
+    """
+    Convert the Execution Trace comms metadata to the common trace format for replay.
+    """
+    ET_PG_NAME_TUPLE = in_trace.schema_pytorch() >= (1, 0, 3)
+    if (in_trace.schema_pytorch() < (1, 0, 3)):
+        raise ValueError(f"Only support trace version >1.0.3, but current trace version is {in_trace.schema.split('-')[0]}")
+
+    pg_ranks_map = _parse_proc_group_info(in_trace) # key is pg id, value is global ranks in this pg
+    comms_op_list = _parse_comms_op_node(in_trace, pg_ranks_map, target_rank, total_ranks)
+
+    return comms_op_list
+
+def _parse_proc_group_info(in_trace: ExecutionTrace):
+    pg_ranks_map = {} # {node_id : {process_group_id : [ranks] } }
+    for node in in_trace.nodes.values():
```

Comment:
nit stylistic change, you can use a generator
pg_init_nodes = (node for node in in_trace.nodes if if "process_group:init" in node.name)

for node in pg_init_nodes:
   ...

Reply:
Done. Thanks!